src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +529,336 @@ private static String trimExcessWhitespaces(String text) {
         return result.toString();
     }
 
+    /**
+     * Finds and return if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        boolean contains = false;
+        for (DetailNode node : javadoc.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = node.getChildren();
+
+                // Checking size of child is not required, since child contains
+                // children of Inline Tag, as at least 2 children will be present
+                // which are RCURLY and LCURLY.
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                    && SUMMARY_TEXT.equals(child[1].getText())) {
+                    contains = true;
+                }
+            }
+        }
+
+        if (containsSummaryTagWithHtmlTag(javadoc)) {
+            contains = true;
+        }
+
+        return contains;
+    }
+
+    /**
+     * Finds and return if summary tag present as HTML Format.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTagWithHtmlTag(DetailNode javadoc) {
+        boolean contains = false;
+        if (getInlineTagNodeInsideHtml(javadoc) != null) {
+            final DetailNode[] htmlChild = getInlineTagNodeInsideHtml(javadoc).getChildren();
```

Comment:
```
inlineTagNode = getInlineTagNodeInsideHtml(javadoc)
        if (inlineTagNode != null) {
            final DetailNode[] htmlChild = inlineTagNode.getChildren();
```

please apply the same approach to all other executions of getInlineTagNodeInsideHtml

Reply:
Done